### PR TITLE
Fix: RPC nodes form dialog

### PR DESCRIPTION
**Pull request by tewshi**
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: frontend/app/src/composables/settings/general/rpc/form.ts [hunk 1]
```diff
@@ -1 +1,12 @@
-export const useEvmRpcNodeForm = createSharedComposable(useForm<boolean>);
+import { type Blockchain } from '@rotki/common/lib/blockchain';
+import { type MaybeRef } from '@vueuse/shared';
+
+const rpcForms: Record<Blockchain, () => ReturnType<typeof useForm<boolean>>> =
+  {} as Record<Blockchain, () => ReturnType<typeof useForm<boolean>>>;
+
+export const useEvmRpcNodeForm = (chain: MaybeRef<Blockchain>) => {
+  if (!rpcForms[get(chain)]) {
```
Comment (lukicenturi) [inline frontend/app/src/composables/settings/general/rpc/form.ts:8]: put `get(chain)` on a variable

**Review by lukicenturi — APPROVED** [review]
It fixes the problem we had. But the forms composable never removed even when the component is onmounted.

**Comment by kelsos** [comment]
> It fixes the problem we had. But the forms composable never removed even when the component is onmounted.

That might be a problem, any way to track the scope and dispose on unmount?

**Comment by tewshi** [comment]
was going to comment, but I see @kelsos has already handled... thanks.

**Review by lukicenturi — APPROVED** [review]
LGTM!

File: frontend/app/src/components/settings/general/rpc/EvmRpcNodeManager.vue [hunk 2]
```diff
@@ -6,6 +6,7 @@ import {
   type EvmRpcNodeList,
   getPlaceholderNode
 } from '@/types/settings';
+import { disposeEvmRpcNodeComposables } from '@/composables/settings/general/rpc/form';
```
Comment (lukicenturi) [inline frontend/app/src/components/settings/general/rpc/EvmRpcNodeManager.vue:9]: this can be removed right?

**Review by kelsos — APPROVED** [review]

**Pull request merged**